main.py

```diff
@@ -826,96 +831,282 @@ def on_link_down(self, event):
+
+            # Swap from current path to failover path
+
+            if swap_to_failover:
+                success, failure = self.execute_swap_to_failover(
+                    swap_to_failover
                 )
-                if offset is None or offset >= len(switch_flows[dpid]):
-                    del switch_flows[dpid]
-                    continue
-                switch_flows[dpid] = switch_flows[dpid][offset:]
-            time.sleep(settings.BATCH_INTERVAL)
 
-        for evc in evcs_normal:
-            emit_event(
-                self.controller,
-                "evc_affected_by_link_down",
-                content={"link": link} | map_evc_event_content(evc)
-            )
+                clear_failover.extend(success)
 
-        evcs_to_update = []
-        for evc in evcs_with_failover:
-            evcs_to_update.append(evc.as_dict())
-            emit_event(
-                self.controller,
-                "redeployed_link_down",
-                content=map_evc_event_content(evc)
-            )
-            log.info(
-                f"{evc} redeployed with failover due to link down {link.id}"
-            )
-        for evc in check_failover:
-            evcs_to_update.append(evc.as_dict())
+                evcs_to_update.update((evc.id, evc) for evc in success)
 
-        self.mongo_controller.update_evcs(evcs_to_update)
+                undeploy.extend(failure)
 
-        emit_event(
-            self.controller,
-            "cleanup_evcs_old_path",
-            content={"evcs": evcs_with_failover + check_failover}
-        )
+            # Clear out failover path
+
+            if clear_failover:
+                success, failure = self.execute_clear_failover(clear_failover)
+
+                evcs_to_update.update((evc.id, evc) for evc in success)
+
+                undeploy.extend(failure)
+
+            # Undeploy the evc, schedule a redeploy
+
+            if undeploy:
+                success, failure = self.execute_undeploy(undeploy)
+
+                evcs_to_update.update((evc.id, evc) for evc in success)
+
+                if failure:
+                    log.error(f"Failed to handle_link_down for {failure}")
+
+            # Push update to DB
+
+            if evcs_to_update:
+                self.mongo_controller.update_evcs(
+                    [evc.as_dict() for evc in evcs_to_update.values()]
+                )
+
+    @listen_to("kytos/mef_eline.need_redeploy")
+    def on_evc_need_redeploy(self, event):
+        """Redeploy evcs that need to be redeployed."""
+        self.handle_evc_need_redeploy(event)
+
+    def handle_evc_need_redeploy(self, event):
+        """Redeploy evcs that need to be redeployed."""
+        evc = self.circuits.get(event.content["evc_id"])
+        if evc is None:
+            return
+        with evc.lock:
+            if not evc.is_enabled() or evc.is_active():
+                return
+            result = evc.deploy_to_path()
+            # if result:
```

Comment:
If you can clean up this too, and review if everything else was copied correctly